components/omega/doc/design/OmegaDesignTemplate.md

```diff
@@ -1,5 +1,5 @@
-# OMEGA Requirements and Design:
-## *MyClassOrModuleName*
+(omega-dsgn-my-class)=
+# Template: MyClassOrModuleName
```

Comment:
@philipwjones, please let me know if you're okay with the changes to the design-doc template.  The existing title formatting isn't working well in the docs:
![image](https://github.com/E3SM-Project/Omega/assets/4179064/5535d55b-32c6-4c04-9e61-6d0efcf3e2c8)

I'll clean up the existing design docs in another PR.

Reply:
Here's what that looks like with the clean-up in #15:
![image](https://github.com/E3SM-Project/Omega/assets/4179064/d6d3a865-1d7f-489c-887a-f8b0397f96cc)